lib/auth/streamingV4/trailingChecksumTransform.js

```diff
@@ -0,0 +1,92 @@
+        super({});
+        this.log = log;
+        this.errCb = errCb;
+        this.chunkSizeBuffer = Buffer.alloc(0);
+        this.outputBuffer = Buffer.alloc(0);
+        this.toFlush = Buffer.alloc(0);
+        this.bytesToRead = 0; // when a chunk is advertised, the size is put here and we forward all bytes
+        this.streamClosed = false;
+    }
+
+    /**
+     * This function will remove the trailing checksum from the stream
+     *
+     * @param {Buffer} chunkInput - chunk from request body
+     * @param {string} encoding - Data encoding
+     * @param {function} callback - Callback(err, justDataChunk, encoding)
+     * @return {function} executes callback with err if applicable
+     */
+    _transform(chunkInput, encoding, callback) {
+        let chunk = chunkInput;
+        for (;;) {
+            if (chunk.byteLength === 0) {
+                break;
+            }
+            if (this.streamClosed) {
+                return callback(null, '', encoding);
+            }
+            if (this.bytesToRead > 0) {
+                if (chunk.byteLength <= this.bytesToRead) {
+                    // chunk is smaller than the advertised size
+                    // forward the whole chunk
+                    this.bytesToRead -= chunk.byteLength;
+                    this.outputBuffer = Buffer.concat([this.outputBuffer, chunk]);
+                    break;
+                }
+                // chunk is bigger than the advertised size
+                this.outputBuffer = Buffer.concat([this.outputBuffer, chunk.subarray(0, this.bytesToRead)]);
+                chunk = chunk.subarray(this.bytesToRead);
+                this.bytesToRead = 0;
+                continue;
+            }
+            const lineBreakIndex = chunk.indexOf('\r\n');
+            if (lineBreakIndex === -1) {
+                this.chunkSizeBuffer = Buffer.concat([this.chunkSizeBuffer, chunk]);
+                if (this.chunkSizeBuffer.byteLength > 8) {
+                    // if bigger, the chunk would be over 5 GB
+                    // returning early to avoid a DoS by memory exhaustion
+                    return callback(errors.InvalidArgument);
+                }
+                return callback(null, '', encoding);
+            }
+            if (lineBreakIndex === 0) {
+                chunk = chunk.subarray(lineBreakIndex + 2);
+                continue;
+            }
+            this.chunkSizeBuffer = Buffer.concat([this.chunkSizeBuffer, chunk.subarray(0, lineBreakIndex)]);
+            chunk = chunk.subarray(lineBreakIndex + 2);
+            // chunk-size is sent in hex
+            const dataSize = Number.parseInt(this.chunkSizeBuffer.toString(), 16);
+            if (Number.isNaN(dataSize)) {
+                return callback(errors.InvalidArgument);
+            }
+            this.chunkSizeBuffer = Buffer.alloc(0);
+            if (dataSize === 0) {
+                // last chunk, no more data to read, the stream is closed
```

Comment:
I would add a TODO to implement parsing and checking the trailing checksum here.